globus_cli/services/transfer/task/cancel.py

```diff
@@ -10,16 +11,57 @@
 @click.command('cancel', short_help='Cancel a Task',
                help='Cancel a Task owned by the current user')
 @common_options
-@task_id_option(helptext='ID of the Task which you want to cancel')
-def cancel_task(task_id):
+@task_id_option(helptext='ID of the Task which you want to cancel',
+                required=False)
+@click.option('--all', '-a', is_flag=True,
+              help='Cancel all in-progress tasks that you own')
+def cancel_task(all, task_id):
     """
     Executor for `globus transfer task cancel`
     """
+
+    if bool(all) + bool(task_id) != 1:
+        raise click.UsageError('You must pass EITHER the special --all flag '
+                               'to cancel all in-progress tasks OR a single '
+                               'task ID to cancel.')
+
     client = get_client()
 
-    res = client.cancel_task(task_id)
+    if all:
+        from sys import maxsize
+        task_ids = [
+            task_row['task_id']
+            for task_row in client.task_list(
+                filter='type:TRANSFER,DELETE/status:ACTIVE,INACTIVE',
+                fields='task_id',
+                num_results=maxsize,  # FIXME want to ask for "unlimited" set
+            )
+        ]
+
+        if not task_ids:
+            raise click.ClickException('You have no in-progress tasks.')
+
+        if outformat_is_json():
+            safeprint(json.dumps(
+                {
+                    'results': [client.cancel_task(i).data for i in task_ids],
+                    'task_ids': task_ids,
+                },
+                indent=2,
+            ))
+
+        else:
+            task_count = len(task_ids)
+            safeprint('Canceling all tasks ({} total)...'.format(task_count))
+            for task_number, task_id in enumerate(task_ids, start=1):
+                safeprint('{} ({} of {}): {}'.
```

Comment:
This kind of streaming of success output is a generally applicable pattern. Do we want to take any pains to ensure that `safeprint` (perhaps optionally) does unbuffered writes?